SpatialGDK/Source/SpatialGDKServices/Public/LocalReceptionistProxyServerManager.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogLocalReceptionistProxyServerManager, Log, All);
+
+class FLocalReceptionistProxyServerManager
+{
+public:
+	FLocalReceptionistProxyServerManager();
+
+	bool CheckIfPortIsBound();
+	bool KillBlockingPortProcess();
+	bool LocalReceptionistProxyServerPreRunChecks();
+
+	FString GetProcessName();
+
+	void SPATIALGDKSERVICES_API Init();
+	bool SPATIALGDKSERVICES_API TryStartReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName);
+	bool SPATIALGDKSERVICES_API TryStopReceptionistProxyServer();
+	
+private:
+
+	struct BlockingPortProcess
+	{
+		FString Pid = "";
+		FString Name = "";
+	};
+
+	BlockingPortProcess BlockingProcess;
+
+	FProcHandle ProxyServerProcHandle;
+	FString RunningCloudDeploymentName;
+
+	static const int32 ExitCodeSuccess = 0;
+	static const int32 ReceptionistPort = 7777;
```

Comment:
checking the spatial cloud connect external command, I just saw that you can specify a different port for the receptionist via `--local_receptionist_port`. maybe use this as the default one and make it configurable in the settings?

Reply:
we also have a `--listening_address`.  might be good to expose that as well?

Reply:
Yep sure I had a related question : https://github.com/spatialos/UnrealGDK/pull/2180#discussion_r436649941. I am going to add it :)